kustomize/README.md

```diff
@@ -0,0 +1,228 @@
+   terrascan server for remote repository scan.
+
+   Note: Before running the command, please verify once that the `server/kustomization.yaml` is set with the desired parameters.
+
+    ```bash
+    kustomize build server/ | kubectl apply -f -
+    ```
+
+### Deploying Terrascan Server in TLS Mode
+
+Deploy terrascan in server mode operating in HTTPS mode.
+
+1. Follow Step 1 from `Deploying Terrascan Server` section
+
+2. Create a domain.cnf file.
+
+    ```bash
+    mkdir server-tls/certs
+    touch server-tls/certs/domain.cnf
+    cat << EOF > certs/domain.cnf
+    [req]
+    default_bits = 2048
+    prompt = no
+    default_md = sha256
+    x509_extensions = v3_req
+    distinguished_name = dn
+    [dn]
+    C = <My_Country>
+    ST = <My_State>
+    L = <My_Location>
+    O = <My_Organization>
+    emailAddress = <My_Email>
+    CN = terrascan.<TERRASCAN_NAMESPACE>.svc.cluster.local
+    [v3_req]
+    subjectAltName = @alt_names
+    [alt_names]
+    DNS.1 = terrascan.<TERRASCAN_NAMESPACE>.svc.cluster.local
+    >EOF
+    ```
+
+    **Note:** Please replace the placeholders like `<My_Country>`,`<My_State>` etc as per your requirements.
+
+3. Generate `server.key` and `server.crt`.
+
+    ```bash
+    openssl req -x509 -sha256 -nodes -newkey rsa:2048 -keyout server-tls/certs/server.key -out server-tls/certs/server.crt -config server-tls/certs/domain.cnf
+    ```
+
+4. Deploy the terrascan server. Skip this step if you're aiming to setup terrascan webhook or terrascan server for remote repository scan.
+
+   **Note:** Before running the command, please verify once that the `server-tls/kustomization.yaml` is set with the desired parameters.
+
+    ```bash
+    kustomize build server-tls/ | kubectl apply -f -
+    ```
+
+### Deploying Terrascan Server For Remote Repository Scan
+
+For scanning remote IaC file repositories, Terrascan must be provided with the required SSH keys to connect and clone the
+repository locally to scan it. The following steps will help in setting up for that.
+
+1. Follow steps 1-3 of the `Deploying TerraScan Server in TLS mode` section.
+
+2. Generate SSH keys and copy `~/.ssh/config`, `~/.ssh/known-hosts` and `~/.ssh/<generated_private_key>` to
+   `server-remote-repo-scan/.ssh/` directory. Replace `<SSH_KEY_NAME>` with your private ssh key's name in
+   `server-remote-repo-scan/kustomization.yaml` and setup the generated public ssh key on your respective code repository
+   hosting service, like github or bitbucket.
+
+   **You may also use this shell command:**
+
+   _Let's assume your private key is `~/.ssh/github_rsa`_
+
+    ```bash
+    mkdir server-remote-repo-scan/.ssh
+    cp ~/.ssh/config ~/.ssh/known_hosts server-remote-repo-scan/.ssh/
+    cp ~/.ssh/github_rsa ~/.ssh/github_rsa.pub server-remote-repo-scan/.ssh/
+    sed s/<SSH_KEY_NAME>/github_rsa/g server-remote-repo-scan/kustomization.yaml
+    ```
+
+3. Deploy. Skip this step if you're aiming to setup terrascan webhook.
```

Comment:
Can we change this description it is a bit misleading. The main purpose of this is to just set up the terrascan server to scan the remote repo.